packages/core/src/state-management/state-manager.ts

```diff
@@ -75,14 +75,17 @@ export class StateManager {
   /**
    * Takes a stream state that might not contain the complete log (and might in fact contain only the
    * genesis commit) and kicks off the process to load and apply the most recent Tip to it.
-   * @param state$
-   * @param timeoutMillis
+   *
+   * @param state$ - Current stream state.
+   * @param timeoutMillis - How much time do we wait for a response from the network.
+   * @param hint - Tip to try while we are waiting for the network to respond.
    */
-  async sync(state$: RunningState, timeoutMillis: number): Promise<void> {
+  async sync(state$: RunningState, timeoutMillis: number, hint?: CID): Promise<void> {
     const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id)
+    const tipSource$ = hint ? merge(tip$, of(hint)) : tip$
     // We do not expect this promise to return anything, so set `defaultValue` to `undefined`
     await lastValueFrom(
-      tip$.pipe(
+      tipSource$.pipe(
```

Comment:
do we have to worry about any synchronization issues with syncing and applying two different logs to the same state object in parallel? 
 Could one modify the state object out from under the other?  Isn't that the reason why we have the ExecutionQueue serializing log application in other places?

Reply:
Should not be the case, as concatMap AFAIK waits till the execution of the function is over. Getting a demonstration code ready.

Reply:
https://gist.github.com/ukstv/93ff950d370fa501d9ac9f0251c95313

It works. No concurrent calls.